pkg/domain/infra/abi/play.go

```diff
@@ -114,6 +120,64 @@ func k8sName(content []byte, suffix string) string {
 	return hash[0:12] + "-" + suffix
 }
 
+func PrintProgress(ctx context.Context, ic entities.ContainerEngine, w io.Writer) {
+	for filter := range eventFiltersCh {
+		fmt.Println(filter)
+		var eventOptions entities.EventsOptions
+		eventChannel := make(chan *events.Event)
+		errChannel := make(chan error)
+
+		eventOptions.Filter = []string{filter}
+		eventOptions.EventChan = eventChannel
+
+		go printEvent(ctx, eventChannel, errChannel, w)
+
+		go func() {
+			err := ic.Events(ctx, eventOptions)
+			errChannel <- err
+		}()
+	}
+}
+
+func printEvent(ctx context.Context, eventChannel chan *events.Event, errChannel chan error, w io.Writer) {
+	flush := func() {
+		if flusher, ok := w.(http.Flusher); ok {
+			flusher.Flush()
+		}
+	}
+	enc := json.NewEncoder(w)
+	enc.SetEscapeHTML(true)
+	for {
+
+		select {
+		case <-ctx.Done():
```

Comment:
This might be able to be a single select statement with multiple cases and no default case.  Whichever channel returns first, that one's block executes, then the select statement exits and the loop repeats.   The `break` statement would then just be a `continue` I believe